In [3]:
import pandas as pd

In [4]:
df= pd.DataFrame([[1,2,3,4],[5,6,7,8],[9, 10,11,12]],columns=["A","B","C","D"], index=["X","Y","Z"])
df

,A,B,C,D
X,1,2,3,4
Y,5,6,7,8
Z,9,10,11,12


In [5]:
df.tail(2)

,A,B,C,D
Y,5,6,7,8
Z,9,10,11,12


In [6]:
df.index

Index(['X', 'Y', 'Z'], dtype='str')

In [7]:
df.info

<bound method DataFrame.info of    A   B   C   D
X  1   2   3   4
Y  5   6   7   8
Z  9  10  11  12>

In [8]:
df.describe()

,A,B,C,D
count,3.0,3.0,3.0,3.0
mean,5.0,6.0,7.0,8.0
std,4.0,4.0,4.0,4.0
min,1.0,2.0,3.0,4.0
25%,3.0,4.0,5.0,6.0
50%,5.0,6.0,7.0,8.0
75%,7.0,8.0,9.0,10.0
max,9.0,10.0,11.0,12.0


In [9]:
df["B"].unique()

array([ 2,  6, 10])

In [10]:
## savoir combine de ligne et de colonne.
df.shape

(3, 4)

In [11]:
### lire un ficher csv.
coffes=pd.read_csv("./data/coffee.csv")
coffes

,Day,Coffee Type,Units Sold
0,Monday,Espresso,25
1,Monday,Latte,15
2,Tuesday,Espresso,30
3,Tuesday,Latte,20
4,Wednesday,Espresso,35
5,Wednesday,Latte,25
6,Thursday,Espresso,40
7,Thursday,Latte,30
8,Friday,Espresso,45
9,Friday,Latte,35


In [12]:
## Autre manière de lire le fichier csv par le lien.
coffes1=pd.read_csv("https://raw.githubusercontent.com/KeithGalli/complete-pandas-tutorial/refs/heads/master/warmup-data/coffee.csv")
coffes1

,Day,Coffee Type,Units Sold
0,Monday,Espresso,25
1,Monday,Latte,15
2,Tuesday,Espresso,30
3,Tuesday,Latte,20
4,Wednesday,Espresso,35
5,Wednesday,Latte,25
6,Thursday,Espresso,40
7,Thursday,Latte,30
8,Friday,Espresso,45
9,Friday,Latte,35


In [13]:
coffes1.head(6)

,Day,Coffee Type,Units Sold
0,Monday,Espresso,25
1,Monday,Latte,15
2,Tuesday,Espresso,30
3,Tuesday,Latte,20
4,Wednesday,Espresso,35
5,Wednesday,Latte,25


In [14]:
results=pd.read_csv("./data/results.csv")
results

,year,type,discipline,event,as,athlete_id,noc,team,place,tied,medal
0,1912.0,Summer,Tennis,"Singles, Men (Olympic)",Jean-François Blanchy,1,FRA,NaN,17.0,True,NaN
1,1912.0,Summer,Tennis,"Doubles, Men (Olympic)",Jean-François Blanchy,1,FRA,Jean Montariol,NaN,False,NaN
2,1920.0,Summer,Tennis,"Singles, Men (Olympic)",Jean-François Blanchy,1,FRA,NaN,32.0,True,NaN
3,1920.0,Summer,Tennis,"Doubles, Mixed (Olympic)",Jean-François Blanchy,1,FRA,Jeanne Vaussard,8.0,True,NaN
4,1920.0,Summer,Tennis,"Doubles, Men (Olympic)",Jean-François Blanchy,1,FRA,Jacques Brugnon,4.0,False,NaN
...,...,...,...,...,...,...,...,...,...,...,...
308403,2022.0,Winter,Luge,"Singles, Men (Olympic)",Marián Skupek,148983,SVK,NaN,26.0,False,NaN
308404,2022.0,Winter,Alpine Skiing (Skiing),"Slalom, Women (Olympic)",Elsa Fermbäck,148984,SWE,NaN,28.0,False,NaN
308405,2022.0,Winter,Alpine Skiing (Skiing),"Team, Mixed (Olympic)",Hilma Lövblom,148985,SWE,Sweden,13.0,False,NaN
308406,2022.0,Winter,Alpine Skiing (Skiing),"Giant Slalom, Women (Olympic)",Hilma Lövblom,148985,SWE,NaN,NaN,False,NaN


In [15]:
olympic_data=pd.read_excel("data/olympics-data.xlsx")
olympic_data


,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25
...,...,...,...,...,...,...,...,...,...,...
145495,149222,Polina Luchnikova,2002-01-30,Serov,Sverdlovsk,RUS,ROC,167.0,61.0,NaN
145496,149223,Valeriya Merkusheva,1999-09-20,Moskva (Moscow),Moskva,RUS,ROC,168.0,65.0,NaN
145497,149224,Yuliya Smirnova,1998-05-08,Kotlas,Arkhangelsk,RUS,ROC,163.0,55.0,NaN
145498,149225,André Foussard,1899-05-19,Niort,Deux-Sèvres,FRA,France,166.0,NaN,1986-03-18


In [16]:
bios=pd.read_csv("./data/bios.csv")
bios

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25
...,...,...,...,...,...,...,...,...,...,...
145495,149222,Polina Luchnikova,2002-01-30,Serov,Sverdlovsk,RUS,ROC,167.0,61.0,NaN
145496,149223,Valeriya Merkusheva,1999-09-20,Moskva (Moscow),Moskva,RUS,ROC,168.0,65.0,NaN
145497,149224,Yuliya Smirnova,1998-05-08,Kotlas,Arkhangelsk,RUS,ROC,163.0,55.0,NaN
145498,149225,André Foussard,1899-05-19,Niort,Deux-Sèvres,FRA,France,166.0,NaN,1986-03-18


In [17]:
coffes

,Day,Coffee Type,Units Sold
0,Monday,Espresso,25
1,Monday,Latte,15
2,Tuesday,Espresso,30
3,Tuesday,Latte,20
4,Wednesday,Espresso,35
5,Wednesday,Latte,25
6,Thursday,Espresso,40
7,Thursday,Latte,30
8,Friday,Espresso,45
9,Friday,Latte,35


In [18]:
coffes.iloc[[1,2,3]]

,Day,Coffee Type,Units Sold
1,Monday,Latte,15
2,Tuesday,Espresso,30
3,Tuesday,Latte,20


FILTERING DATA


In [19]:
bios

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25
...,...,...,...,...,...,...,...,...,...,...
145495,149222,Polina Luchnikova,2002-01-30,Serov,Sverdlovsk,RUS,ROC,167.0,61.0,NaN
145496,149223,Valeriya Merkusheva,1999-09-20,Moskva (Moscow),Moskva,RUS,ROC,168.0,65.0,NaN
145497,149224,Yuliya Smirnova,1998-05-08,Kotlas,Arkhangelsk,RUS,ROC,163.0,55.0,NaN
145498,149225,André Foussard,1899-05-19,Niort,Deux-Sèvres,FRA,France,166.0,NaN,1986-03-18


In [20]:
### qarde juste les personne né au usa , france , et grande bretagne.
bios[bios["born_country"].isin(["USA","FRA","GBR"])]


,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25
...,...,...,...,...,...,...,...,...,...,...
145454,149180,Anna Hoffmann,2000-03-28,Madison,Wisconsin,USA,United States,NaN,NaN,NaN
145457,149183,Alix Wilkinson,2000-08-02,Mammoth Lakes,California,USA,United States,NaN,NaN,NaN
145467,149194,Elisa Nakab,1998-05-08,Briançon,Hautes-Alpes,FRA,Italy,NaN,NaN,NaN
145468,149195,Justin Abdelkader,1987-02-25,Muskegon,Michigan,USA,United States,187.0,97.0,NaN


Addiction et retirer des colonnes


In [21]:
coffes['price'] = 4.99
coffes

,Day,Coffee Type,Units Sold,price
0,Monday,Espresso,25,4.99
1,Monday,Latte,15,4.99
2,Tuesday,Espresso,30,4.99
3,Tuesday,Latte,20,4.99
4,Wednesday,Espresso,35,4.99
5,Wednesday,Latte,25,4.99
6,Thursday,Espresso,40,4.99
7,Thursday,Latte,30,4.99
8,Friday,Espresso,45,4.99
9,Friday,Latte,35,4.99


In [22]:
bios

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25
...,...,...,...,...,...,...,...,...,...,...
145495,149222,Polina Luchnikova,2002-01-30,Serov,Sverdlovsk,RUS,ROC,167.0,61.0,NaN
145496,149223,Valeriya Merkusheva,1999-09-20,Moskva (Moscow),Moskva,RUS,ROC,168.0,65.0,NaN
145497,149224,Yuliya Smirnova,1998-05-08,Kotlas,Arkhangelsk,RUS,ROC,163.0,55.0,NaN
145498,149225,André Foussard,1899-05-19,Niort,Deux-Sèvres,FRA,France,166.0,NaN,1986-03-18


In [23]:
coffes

,Day,Coffee Type,Units Sold,price
0,Monday,Espresso,25,4.99
1,Monday,Latte,15,4.99
2,Tuesday,Espresso,30,4.99
3,Tuesday,Latte,20,4.99
4,Wednesday,Espresso,35,4.99
5,Wednesday,Latte,25,4.99
6,Thursday,Espresso,40,4.99
7,Thursday,Latte,30,4.99
8,Friday,Espresso,45,4.99
9,Friday,Latte,35,4.99


In [24]:
import numpy as np

In [25]:
### suprimer une colonne.
coffes.drop(columns=['Day'])
coffes

,Day,Coffee Type,Units Sold,price
0,Monday,Espresso,25,4.99
1,Monday,Latte,15,4.99
2,Tuesday,Espresso,30,4.99
3,Tuesday,Latte,20,4.99
4,Wednesday,Espresso,35,4.99
5,Wednesday,Latte,25,4.99
6,Thursday,Espresso,40,4.99
7,Thursday,Latte,30,4.99
8,Friday,Espresso,45,4.99
9,Friday,Latte,35,4.99


In [26]:
coffes["revenu"] = coffes['Units Sold']*coffes['price']
coffes

,Day,Coffee Type,Units Sold,price,revenu
0,Monday,Espresso,25,4.99,124.75
1,Monday,Latte,15,4.99,74.85
2,Tuesday,Espresso,30,4.99,149.70
3,Tuesday,Latte,20,4.99,99.80
4,Wednesday,Espresso,35,4.99,174.65
5,Wednesday,Latte,25,4.99,124.75
6,Thursday,Espresso,40,4.99,199.60
7,Thursday,Latte,30,4.99,149.70
8,Friday,Espresso,45,4.99,224.55
9,Friday,Latte,35,4.99,174.65


In [27]:
##" renormé une colonne"
coffes.rename(columns = { 'price' : 'new_price'})

,Day,Coffee Type,Units Sold,new_price,revenu
0,Monday,Espresso,25,4.99,124.75
1,Monday,Latte,15,4.99,74.85
2,Tuesday,Espresso,30,4.99,149.70
3,Tuesday,Latte,20,4.99,99.80
4,Wednesday,Espresso,35,4.99,174.65
5,Wednesday,Latte,25,4.99,124.75
6,Thursday,Espresso,40,4.99,199.60
7,Thursday,Latte,30,4.99,149.70
8,Friday,Espresso,45,4.99,224.55
9,Friday,Latte,35,4.99,174.65


In [28]:
print(coffes.columns)

Index(['Day', 'Coffee Type', 'Units Sold', 'price', 'revenu'], dtype='str')


In [29]:
coffes.rename(columns = {'Day' : "jour",
                         'Coffee Type': 'Type café',
                         'revenu' : "Revenue",
                         'Units Sold' : 'Unité vendue'})

,jour,Type café,Unité vendue,price,Revenue
0,Monday,Espresso,25,4.99,124.75
1,Monday,Latte,15,4.99,74.85
2,Tuesday,Espresso,30,4.99,149.70
3,Tuesday,Latte,20,4.99,99.80
4,Wednesday,Espresso,35,4.99,174.65
5,Wednesday,Latte,25,4.99,124.75
6,Thursday,Espresso,40,4.99,199.60
7,Thursday,Latte,30,4.99,149.70
8,Friday,Espresso,45,4.99,224.55
9,Friday,Latte,35,4.99,174.65


In [30]:
coffes.rename(columns = {'Day' : "jour",
                         'Coffee Type': 'Type café',
                         'revenu' : "Revenue",
                         'Units Sold' : 'Unité vendue'}, inplace  = True)

In [31]:
coffes

,jour,Type café,Unité vendue,price,Revenue
0,Monday,Espresso,25,4.99,124.75
1,Monday,Latte,15,4.99,74.85
2,Tuesday,Espresso,30,4.99,149.70
3,Tuesday,Latte,20,4.99,99.80
4,Wednesday,Espresso,35,4.99,174.65
5,Wednesday,Latte,25,4.99,124.75
6,Thursday,Espresso,40,4.99,199.60
7,Thursday,Latte,30,4.99,149.70
8,Friday,Espresso,45,4.99,224.55
9,Friday,Latte,35,4.99,174.65


In [32]:
bios.head(10)

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25
5,6,Nicolas Chatelain,1970-01-13,Amiens,Somme,FRA,France,181.0,70.0,NaN
6,7,Patrick Chila,1969-11-27,Ris-Orangis,Essonne,FRA,France,180.0,73.0,NaN
7,8,Henri Cochet,1901-12-14,Villeurbanne,Rhône,FRA,France,NaN,NaN,1987-04-02
8,9,Marcel Cousin,1896-08-04,Nîmes,Gard,FRA,France,NaN,NaN,1986-08-01
9,10,Guy de la Chapelle,1868-07-16,Farges-Allichamps,Cher,FRA,France,NaN,NaN,1923-08-27


In [33]:
Bios_french = bios.rename(columns = { 'name' : 'nom',
                              'born_date' : 'Date_de_naissance',
                              'born-city' : 'Ville_de_naissance', 
                              'born-region' : 'Région_de_naissance'})
Bios_french

,athlete_id,nom,Date_de_naissance,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25
...,...,...,...,...,...,...,...,...,...,...
145495,149222,Polina Luchnikova,2002-01-30,Serov,Sverdlovsk,RUS,ROC,167.0,61.0,NaN
145496,149223,Valeriya Merkusheva,1999-09-20,Moskva (Moscow),Moskva,RUS,ROC,168.0,65.0,NaN
145497,149224,Yuliya Smirnova,1998-05-08,Kotlas,Arkhangelsk,RUS,ROC,163.0,55.0,NaN
145498,149225,André Foussard,1899-05-19,Niort,Deux-Sèvres,FRA,France,166.0,NaN,1986-03-18


In [34]:
new_bios= bios.copy()
new_bios

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25
...,...,...,...,...,...,...,...,...,...,...
145495,149222,Polina Luchnikova,2002-01-30,Serov,Sverdlovsk,RUS,ROC,167.0,61.0,NaN
145496,149223,Valeriya Merkusheva,1999-09-20,Moskva (Moscow),Moskva,RUS,ROC,168.0,65.0,NaN
145497,149224,Yuliya Smirnova,1998-05-08,Kotlas,Arkhangelsk,RUS,ROC,163.0,55.0,NaN
145498,149225,André Foussard,1899-05-19,Niort,Deux-Sèvres,FRA,France,166.0,NaN,1986-03-18


In [35]:
new_bios['first_name'] = new_bios['name'].str.split(' ').str[0]
new_bios

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,first_name
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Jean-François
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN,Arnaud
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17,Jean
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20,Jacques
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25,Albert
...,...,...,...,...,...,...,...,...,...,...,...
145495,149222,Polina Luchnikova,2002-01-30,Serov,Sverdlovsk,RUS,ROC,167.0,61.0,NaN,Polina
145496,149223,Valeriya Merkusheva,1999-09-20,Moskva (Moscow),Moskva,RUS,ROC,168.0,65.0,NaN,Valeriya
145497,149224,Yuliya Smirnova,1998-05-08,Kotlas,Arkhangelsk,RUS,ROC,163.0,55.0,NaN,Yuliya
145498,149225,André Foussard,1899-05-19,Niort,Deux-Sèvres,FRA,France,166.0,NaN,1986-03-18,André


In [36]:
new_bios.info()

<class 'pandas.DataFrame'>
RangeIndex: 145500 entries, 0 to 145499
Data columns (total 11 columns):
 #   Column        Non-Null Count   Dtype  
---  ------        --------------   -----  
 0   athlete_id    145500 non-null  int64  
 1   name          145500 non-null  str    
 2   born_date     143693 non-null  str    
 3   born_city     110908 non-null  str    
 4   born_region   110908 non-null  str    
 5   born_country  110908 non-null  str    
 6   NOC           145499 non-null  str    
 7   height_cm     106651 non-null  float64
 8   weight_kg     102070 non-null  float64
 9   died_date     33940 non-null   str    
 10  first_name    145500 non-null  object 
dtypes: float64(2), int64(1), object(1), str(7)
memory usage: 19.6+ MB


In [37]:
### recrire une colonne anné.
new_bios['born_datetime'] = pd.to_datetime(new_bios['born_date'])
new_bios

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,first_name,born_datetime
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Jean-François,1886-12-12
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN,Arnaud,1969-04-01
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17,Jean,1898-08-13
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20,Jacques,1895-05-11
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25,Albert,1878-04-17
...,...,...,...,...,...,...,...,...,...,...,...,...
145495,149222,Polina Luchnikova,2002-01-30,Serov,Sverdlovsk,RUS,ROC,167.0,61.0,NaN,Polina,2002-01-30
145496,149223,Valeriya Merkusheva,1999-09-20,Moskva (Moscow),Moskva,RUS,ROC,168.0,65.0,NaN,Valeriya,1999-09-20
145497,149224,Yuliya Smirnova,1998-05-08,Kotlas,Arkhangelsk,RUS,ROC,163.0,55.0,NaN,Yuliya,1998-05-08
145498,149225,André Foussard,1899-05-19,Niort,Deux-Sèvres,FRA,France,166.0,NaN,1986-03-18,André,1899-05-19


In [38]:
new_bios["born_year"] = new_bios['born_datetime'].dt.year
new_bios

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,first_name,born_datetime,born_year
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Jean-François,1886-12-12,1886.0
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN,Arnaud,1969-04-01,1969.0
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17,Jean,1898-08-13,1898.0
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20,Jacques,1895-05-11,1895.0
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25,Albert,1878-04-17,1878.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...
145495,149222,Polina Luchnikova,2002-01-30,Serov,Sverdlovsk,RUS,ROC,167.0,61.0,NaN,Polina,2002-01-30,2002.0
145496,149223,Valeriya Merkusheva,1999-09-20,Moskva (Moscow),Moskva,RUS,ROC,168.0,65.0,NaN,Valeriya,1999-09-20,1999.0
145497,149224,Yuliya Smirnova,1998-05-08,Kotlas,Arkhangelsk,RUS,ROC,163.0,55.0,NaN,Yuliya,1998-05-08,1998.0
145498,149225,André Foussard,1899-05-19,Niort,Deux-Sèvres,FRA,France,166.0,NaN,1986-03-18,André,1899-05-19,1899.0


In [39]:
new_bios

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,first_name,born_datetime,born_year
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Jean-François,1886-12-12,1886.0
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN,Arnaud,1969-04-01,1969.0
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17,Jean,1898-08-13,1898.0
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20,Jacques,1895-05-11,1895.0
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25,Albert,1878-04-17,1878.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...
145495,149222,Polina Luchnikova,2002-01-30,Serov,Sverdlovsk,RUS,ROC,167.0,61.0,NaN,Polina,2002-01-30,2002.0
145496,149223,Valeriya Merkusheva,1999-09-20,Moskva (Moscow),Moskva,RUS,ROC,168.0,65.0,NaN,Valeriya,1999-09-20,1999.0
145497,149224,Yuliya Smirnova,1998-05-08,Kotlas,Arkhangelsk,RUS,ROC,163.0,55.0,NaN,Yuliya,1998-05-08,1998.0
145498,149225,André Foussard,1899-05-19,Niort,Deux-Sèvres,FRA,France,166.0,NaN,1986-03-18,André,1899-05-19,1899.0


In [40]:
new_bios[["name","born_year"]]

,name,born_year
0,Jean-François Blanchy,1886.0
1,Arnaud Boetsch,1969.0
2,Jean Borotra,1898.0
3,Jacques Brugnon,1895.0
4,Albert Canet,1878.0
...,...,...
145495,Polina Luchnikova,2002.0
145496,Valeriya Merkusheva,1999.0
145497,Yuliya Smirnova,1998.0
145498,André Foussard,1899.0


In [41]:
## Enregistrer bios en format csv.
new_bios.to_csv('./data/new_bios.csv',index = False)

In [42]:
bios['heigt_category'] = bios ['height_cm'].apply(lambda x: 'short' if x < 165 
                                                 else ( 'average' if x < 185 
                                                       else 'Tail'))
bios.head()

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,heigt_category
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Tail
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN,average
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17,average
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20,average
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25,Tail


def categorize_athelete(row):
if row['height_cm']<175 and row['weight_kg'] < 70:
    return 'Lightweight'
elif row['height_cm] '] <185 or row['weight_kg'] <= 80 :
    return 'Middleweight'
bios['category'] = bios.apply(categorize_athlete, axis = 1 )

FUSION ET CONCATENATION DES DONNEES. ( mettre une collone a coté de l'autre).

In [43]:
bios.head()

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,heigt_category
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Tail
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN,average
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17,average
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20,average
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25,Tail


In [44]:
nocs = pd.read_csv('./data//noc_regions.csv')
nocs.head()

,NOC,region,notes
0,AFG,Afghanistan,NaN
1,AHO,Curacao,Netherlands Antilles
2,ALB,Albania,NaN
3,ALG,Algeria,NaN
4,AND,Andorra,NaN


In [45]:
new_bios = pd.merge(bios, nocs,left_on= 'born_country', right_on= 'NOC', how ='left')
new_bios

,athlete_id,name,born_date,born_city,born_region,born_country,NOC_x,height_cm,weight_kg,died_date,heigt_category,NOC_y,region,notes
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Tail,FRA,France,NaN
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN,average,FRA,France,NaN
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17,average,FRA,France,NaN
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20,average,FRA,France,NaN
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25,Tail,GBR,UK,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
145495,149222,Polina Luchnikova,2002-01-30,Serov,Sverdlovsk,RUS,ROC,167.0,61.0,NaN,average,RUS,Russia,NaN
145496,149223,Valeriya Merkusheva,1999-09-20,Moskva (Moscow),Moskva,RUS,ROC,168.0,65.0,NaN,average,RUS,Russia,NaN
145497,149224,Yuliya Smirnova,1998-05-08,Kotlas,Arkhangelsk,RUS,ROC,163.0,55.0,NaN,short,RUS,Russia,NaN
145498,149225,André Foussard,1899-05-19,Niort,Deux-Sèvres,FRA,France,166.0,NaN,1986-03-18,average,FRA,France,NaN


In [46]:
new_bios.rename(columns= {'region': 'Pays_de_naissance'},inplace= True)
new_bios

,athlete_id,name,born_date,born_city,born_region,born_country,NOC_x,height_cm,weight_kg,died_date,heigt_category,NOC_y,Pays_de_naissance,notes
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Tail,FRA,France,NaN
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN,average,FRA,France,NaN
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17,average,FRA,France,NaN
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20,average,FRA,France,NaN
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25,Tail,GBR,UK,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
145495,149222,Polina Luchnikova,2002-01-30,Serov,Sverdlovsk,RUS,ROC,167.0,61.0,NaN,average,RUS,Russia,NaN
145496,149223,Valeriya Merkusheva,1999-09-20,Moskva (Moscow),Moskva,RUS,ROC,168.0,65.0,NaN,average,RUS,Russia,NaN
145497,149224,Yuliya Smirnova,1998-05-08,Kotlas,Arkhangelsk,RUS,ROC,163.0,55.0,NaN,short,RUS,Russia,NaN
145498,149225,André Foussard,1899-05-19,Niort,Deux-Sèvres,FRA,France,166.0,NaN,1986-03-18,average,FRA,France,NaN


Affiché juste ces qui sont néé au USA,FRA,GBR

In [47]:
USA = bios[bios['born_country'] == 'USA'].copy()
GBR = bios[bios['born_country'] == 'GBR'].copy()
FRA = bios[bios['born_country'] == 'FRA'].copy()


 Les lignes de GBR et FRA s’ajoutent sous USA verticalement 

In [48]:
new_df = pd.concat([USA,GBR,FRA], axis =0).copy()
new_df

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,heigt_category
54,55,Monique Javer,1967-07-22,Burlingame,California,USA,Great Britain,177.0,64.0,NaN,average
960,964,Xóchitl Escobedo,1968-09-17,West Covina,California,USA,Mexico,170.0,60.0,NaN,average
961,965,Angélica Gavaldón,1973-10-03,El Centro,California,USA,Mexico,160.0,54.0,NaN,short
1231,1238,Bert Schneider,1897-07-01,Cleveland,Ohio,USA,Canada,NaN,NaN,1986-02-20,Tail
1345,1352,Laura Berg,1975-01-06,Santa Fe Springs,California,USA,United States,168.0,61.0,NaN,average
...,...,...,...,...,...,...,...,...,...,...,...
144652,148348,James Clugnet,1996-12-04,Grenoble,Isère,FRA,Great Britain,NaN,NaN,NaN,Tail
145009,148720,Margaux Hackett,1999-06-02,Annecy,Haute-Savoie,FRA,New Zealand,NaN,NaN,NaN,Tail
145281,149002,Elliott Baralo,1998-08-01,Annecy,Haute-Savoie,FRA,Sweden,NaN,NaN,NaN,Tail
145467,149194,Elisa Nakab,1998-05-08,Briançon,Hautes-Alpes,FRA,Italy,NaN,NaN,NaN,Tail


In [49]:
## mettre GBR et FRA a coté droite de USA (1)
new_df = pd.concat([USA,GBR,FRA], axis =1).copy()
new_df

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,...,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,heigt_category
54,55.0,Monique Javer,1967-07-22,Burlingame,California,USA,Great Britain,177.0,64.0,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
960,964.0,Xóchitl Escobedo,1968-09-17,West Covina,California,USA,Mexico,170.0,60.0,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
961,965.0,Angélica Gavaldón,1973-10-03,El Centro,California,USA,Mexico,160.0,54.0,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1231,1238.0,Bert Schneider,1897-07-01,Cleveland,Ohio,USA,Canada,NaN,NaN,1986-02-20,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1345,1352.0,Laura Berg,1975-01-06,Santa Fe Springs,California,USA,United States,168.0,61.0,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
144652,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,James Clugnet,1996-12-04,Grenoble,Isère,FRA,Great Britain,NaN,NaN,NaN,Tail
145009,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,Margaux Hackett,1999-06-02,Annecy,Haute-Savoie,FRA,New Zealand,NaN,NaN,NaN,Tail
145281,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,Elliott Baralo,1998-08-01,Annecy,Haute-Savoie,FRA,Sweden,NaN,NaN,NaN,Tail
145467,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,Elisa Nakab,1998-05-08,Briançon,Hautes-Alpes,FRA,Italy,NaN,NaN,NaN,Tail


In [50]:
new_df.tail()

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,...,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,heigt_category
144652,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,James Clugnet,1996-12-04,Grenoble,Isère,FRA,Great Britain,NaN,NaN,NaN,Tail
145009,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,Margaux Hackett,1999-06-02,Annecy,Haute-Savoie,FRA,New Zealand,NaN,NaN,NaN,Tail
145281,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,Elliott Baralo,1998-08-01,Annecy,Haute-Savoie,FRA,Sweden,NaN,NaN,NaN,Tail
145467,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,Elisa Nakab,1998-05-08,Briançon,Hautes-Alpes,FRA,Italy,NaN,NaN,NaN,Tail
145498,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,André Foussard,1899-05-19,Niort,Deux-Sèvres,FRA,France,166.0,NaN,1986-03-18,average


In [51]:
bios['height_cm'].unique()


array([ nan, 183., 168., 181., 180., 165., 189., 178., 184., 177., 186.,
       175., 176., 190., 157., 173., 170., 172., 171., 185., 191., 182.,
       163., 174., 169., 192., 196., 187., 179., 160., 166., 164., 188.,
       167., 161., 193., 162., 195., 156., 159., 152., 158., 146., 155.,
       150., 154., 200., 208., 194., 199., 153., 204., 198., 203., 197.,
       206., 205., 143., 144., 151., 135., 147., 127., 128., 130., 140.,
       133., 201., 202., 220., 210., 213., 217., 215., 207., 211., 221.,
       214., 218., 209., 223., 212., 216., 145., 142., 148., 138., 136.,
       149., 137., 132., 141., 139., 131., 226., 219.])

In [52]:
## filtrer des personnes  dont leur taille est > 180 et  <156.
Taill_180_156= bios[(bios['height_cm'] >156 ) & (bios['height_cm'] < 180 )].copy()
Taill_180_156

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,heigt_category
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20,average
14,15,Damien Éloi,1969-07-04,Vire,Calvados,FRA,France,165.0,58.0,NaN,average
17,18,Jean-Philippe Gatien,1968-10-16,Alès,Gard,FRA,France,178.0,73.0,NaN,average
26,27,Christophe Legoût,1973-08-06,Montbéliard,Doubs,FRA,France,177.0,75.0,NaN,average
34,35,Fabrice Santoro,1972-12-09,Tahiti,Îles du Vent,FPN,France,177.0,68.0,NaN,average
...,...,...,...,...,...,...,...,...,...,...,...
145474,149201,Nadine Hofstetter,1994-10-21,Romoos,Luzern,SUI,Switzerland,164.0,68.0,NaN,short
145495,149222,Polina Luchnikova,2002-01-30,Serov,Sverdlovsk,RUS,ROC,167.0,61.0,NaN,average
145496,149223,Valeriya Merkusheva,1999-09-20,Moskva (Moscow),Moskva,RUS,ROC,168.0,65.0,NaN,average
145497,149224,Yuliya Smirnova,1998-05-08,Kotlas,Arkhangelsk,RUS,ROC,163.0,55.0,NaN,short


In [53]:
filt = bios[bios['height_cm'].between(156,180)].copy()

In [54]:
new_bios

,athlete_id,name,born_date,born_city,born_region,born_country,NOC_x,height_cm,weight_kg,died_date,heigt_category,NOC_y,Pays_de_naissance,notes
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Tail,FRA,France,NaN
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN,average,FRA,France,NaN
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17,average,FRA,France,NaN
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20,average,FRA,France,NaN
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25,Tail,GBR,UK,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
145495,149222,Polina Luchnikova,2002-01-30,Serov,Sverdlovsk,RUS,ROC,167.0,61.0,NaN,average,RUS,Russia,NaN
145496,149223,Valeriya Merkusheva,1999-09-20,Moskva (Moscow),Moskva,RUS,ROC,168.0,65.0,NaN,average,RUS,Russia,NaN
145497,149224,Yuliya Smirnova,1998-05-08,Kotlas,Arkhangelsk,RUS,ROC,163.0,55.0,NaN,short,RUS,Russia,NaN
145498,149225,André Foussard,1899-05-19,Niort,Deux-Sèvres,FRA,France,166.0,NaN,1986-03-18,average,FRA,France,NaN


In [55]:
### garder toute les ligne ou born_country est USA ou FRANCE OU CANDE autrement
#dit que la colonne pays de naissance vas juste contenir (FRANce, china,Russia).
LES_Pays = new_bios[new_bios['Pays_de_naissance'].isin(['France	','China','Russia','Rwanda'])].copy()
LES_Pays

,athlete_id,name,born_date,born_city,born_region,born_country,NOC_x,height_cm,weight_kg,died_date,heigt_category,NOC_y,Pays_de_naissance,notes
83,84,Xinhua Chen,1960-01-30,Fuan,Fujian,CHN,Great Britain,175.0,71.0,NaN,average,CHN,China,NaN
133,134,Jie Schöpp,1968-01-25,Baoding,Hebei,CHN,Germany,174.0,63.0,NaN,average,CHN,China,NaN
158,159,Robert Spies,1891-02-17,Moskva (Moscow),Moskva,RUS,Germany,NaN,NaN,1982-10-22,Tail,RUS,Russia,NaN
381,382,He Ying,1977-04-17,Siping,Jilin,CHN,People's Republic of China,170.0,68.0,NaN,average,CHN,China,NaN
383,384,Ma Xiangjun,1964-10-30,Qingdao,Shandong,CHN,People's Republic of China,171.0,65.0,NaN,average,CHN,China,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
145487,149214,Yekaterina Kosova,1996-04-25,Moskva (Moscow),Moskva,RUS,ROC,NaN,NaN,NaN,Tail,RUS,Russia,NaN
145493,149220,Landysh Falyakhova,1998-08-31,Dva Polya Artash,Respublika Tatarstan,RUS,ROC,NaN,NaN,NaN,Tail,RUS,Russia,NaN
145495,149222,Polina Luchnikova,2002-01-30,Serov,Sverdlovsk,RUS,ROC,167.0,61.0,NaN,average,RUS,Russia,NaN
145496,149223,Valeriya Merkusheva,1999-09-20,Moskva (Moscow),Moskva,RUS,ROC,168.0,65.0,NaN,average,RUS,Russia,NaN


In [56]:
new_bios.columns

Index(['athlete_id', 'name', 'born_date', 'born_city', 'born_region',
       'born_country', 'NOC_x', 'height_cm', 'weight_kg', 'died_date',
       'heigt_category', 'NOC_y', 'Pays_de_naissance', 'notes'],
      dtype='str')

AUTRE ATERNALTIVE

In [57]:
#str.contains() — filtrer du texte
#Pour filtrer les lignes contenant un mot ou une partie de mot :
filtre_mot=new_bios[new_bios['Pays_de_naissance'].str.contains('ana',case= False,na= False)]


Voire alors les élement qui contient mon filtre.

In [58]:
filtre_mot['Pays_de_naissance'].unique()

<ArrowStringArray>
['Canada', 'Guyana', 'Ghana', 'Panama', 'Botswana']
Length: 5, dtype: str

EXCLURE DES VALEURS(~), afficher d'autre pays autre que France et Usa

In [59]:
filt = new_bios[~new_bios['Pays_de_naissance'].isin(["USA" , "France"])].copy()
filt['Pays_de_naissance'].unique() ## pour montrer que la france et USA ne sont plus dans la colonne.

<ArrowStringArray>
[                     'UK',                       nan,
                 'Morocco',                    'Iraq',
            'South Africa',                 'Ireland',
                   'India',                   'China',
                 'Jamaica',               'Australia',
 ...
                'Maldives',                  'Rwanda',
              'Micronesia',                    'Laos',
                   'Tonga',        'Marshall Islands',
                  'Brunei',              'Mauritania',
            'Cook Islands', 'Virgin Islands, British']
Length: 200, dtype: str

Filter sur plusieurs colonne avec OR (|)

In [60]:
fitr = new_bios[(new_bios['born_city'] == 'China') |
                (new_bios['born_city'] == 'India')].copy() ## affiche les lignes contenant  china ou India
fitr

,athlete_id,name,born_date,born_city,born_region,born_country,NOC_x,height_cm,weight_kg,died_date,heigt_category,NOC_y,Pays_de_naissance,notes
72962,73520,Raúl González,1952-02-29,China,Nuevo León,MEX,Mexico,175.0,64.0,NaN,average,MEX,Mexico,NaN


Filtre sur les valeur manquante. garde les ligne ou la taille est manquante

In [61]:
manque= new_bios[new_bios['height_cm'].isna()].copy()
manque

,athlete_id,name,born_date,born_city,born_region,born_country,NOC_x,height_cm,weight_kg,died_date,heigt_category,NOC_y,Pays_de_naissance,notes
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Tail,FRA,France,NaN
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25,Tail,GBR,UK,NaN
7,8,Henri Cochet,1901-12-14,Villeurbanne,Rhône,FRA,France,NaN,NaN,1987-04-02,Tail,FRA,France,NaN
8,9,Marcel Cousin,1896-08-04,Nîmes,Gard,FRA,France,NaN,NaN,1986-08-01,Tail,FRA,France,NaN
9,10,Guy de la Chapelle,1868-07-16,Farges-Allichamps,Cher,FRA,France,NaN,NaN,1923-08-27,Tail,FRA,France,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
145490,149217,Sin Ye-Chan,1995-06-13,NaN,NaN,NaN,Republic of Korea,NaN,NaN,NaN,Tail,NaN,NaN,NaN
145491,149218,Matthew Wepke,1989-12-05,NaN,NaN,NaN,Jamaica,NaN,NaN,NaN,Tail,NaN,NaN,NaN
145492,149219,Carlos García-Ordóñez,1927-04-24,La Habana (Havana),Ciudad de La Habana,CUB,Cuba,NaN,NaN,2019-11-24,Tail,CUB,Cuba,NaN
145493,149220,Landysh Falyakhova,1998-08-31,Dva Polya Artash,Respublika Tatarstan,RUS,ROC,NaN,NaN,NaN,Tail,RUS,Russia,NaN


In [62]:
new_bios['born_city'].unique()

<ArrowStringArray>
[           'Bordeaux',              'Meulan',            'Biarritz',
         'Paris VIIIe',          'Wandsworth',              'Amiens',
         'Ris-Orangis',        'Villeurbanne',               'Nîmes',
   'Farges-Allichamps',
 ...
 'Grosse Pointe Woods',        'Mount Shasta',             'McHenry',
           'Scandiano',          'Port Moody',            'Tsivilsk',
                'Roth',              'Romoos',    'Dva Polya Artash',
        'Dulwich Hill']
Length: 22369, dtype: str

In [63]:
### garder toute les ligne ou born_country est USA ou FRANCE OU CANDE autrement
#dit que la colonne pays de naissance vas juste contenir (FRANce, china,Russia).
LES_Pays_2 = new_bios[
    (new_bios['Pays_de_naissance'] == 'France') |
    (new_bios['Pays_de_naissance'] == 'Russia') |
    (new_bios['Pays_de_naissance'] == 'China') |
    (new_bios['Pays_de_naissance'] == 'Rwanda')].copy()
LES_Pays_2

,athlete_id,name,born_date,born_city,born_region,born_country,NOC_x,height_cm,weight_kg,died_date,heigt_category,NOC_y,Pays_de_naissance,notes
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Tail,FRA,France,NaN
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN,average,FRA,France,NaN
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17,average,FRA,France,NaN
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20,average,FRA,France,NaN
5,6,Nicolas Chatelain,1970-01-13,Amiens,Somme,FRA,France,181.0,70.0,NaN,average,FRA,France,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
145493,149220,Landysh Falyakhova,1998-08-31,Dva Polya Artash,Respublika Tatarstan,RUS,ROC,NaN,NaN,NaN,Tail,RUS,Russia,NaN
145495,149222,Polina Luchnikova,2002-01-30,Serov,Sverdlovsk,RUS,ROC,167.0,61.0,NaN,average,RUS,Russia,NaN
145496,149223,Valeriya Merkusheva,1999-09-20,Moskva (Moscow),Moskva,RUS,ROC,168.0,65.0,NaN,average,RUS,Russia,NaN
145497,149224,Yuliya Smirnova,1998-05-08,Kotlas,Arkhangelsk,RUS,ROC,163.0,55.0,NaN,short,RUS,Russia,NaN


In [64]:
## savoir les élements qui contienent la colone pays de naissance.
new_bios['Pays_de_naissance'].unique()

<ArrowStringArray>
[                 'France',                      'UK',
                       nan,                 'Morocco',
                    'Iraq',                     'USA',
            'South Africa',                 'Ireland',
                   'India',                   'China',
 ...
                'Maldives',                  'Rwanda',
              'Micronesia',                    'Laos',
                   'Tonga',        'Marshall Islands',
                  'Brunei',              'Mauritania',
            'Cook Islands', 'Virgin Islands, British']
Length: 202, dtype: str

In [65]:
### Comment trouver les element dans cette colonne .
bios['heigt_category'].unique()

<ArrowStringArray>
['Tail', 'average', 'short']
Length: 3, dtype: str

In [66]:
### Comment trouver les element dans cette colonne .
bios['born_region'].unique()

<ArrowStringArray>
[              'Gironde',              'Yvelines',  'Pyrénées-Atlantiques',
                 'Paris',               'England',                 'Somme',
               'Essonne',                 'Rhône',                  'Gard',
                  'Cher',
 ...
          'Panamá Oeste',               'Ucayali',         'Davao del Sur',
 'Ramallah and Al-Bireh',               'Komenda',               'Kalasin',
                 'Iğdır',               'Ağstafa',         'Sihanoukville',
             'Vĩnh Long']
Length: 2555, dtype: str

In [67]:
## Affiché ceux qui  sont à paris et ont une taille Tail.
Paris = bios[(bios['born_region']  == 'Paris') & (bios['heigt_category'] == 'Tail')].copy()
Paris

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,heigt_category
11,12,Max Decugis,1882-09-24,Paris XVIe,Paris,FRA,France,NaN,NaN,1978-09-06,Tail
15,16,Adrien Fauchier-Magnan,1873-11-19,Paris VIIIe,Paris,FRA,France,NaN,NaN,1965-08-06,Tail
19,20,André Gobert,1890-09-30,Paris XVIIe,Paris,FRA,France,NaN,NaN,1951-12-06,Tail
20,21,Pierre Hirsch,1900-11-08,Paris Ie,Paris,FRA,France,NaN,NaN,1942-04-19,Tail
21,22,René Lacoste,1904-07-02,Paris Xe,Paris,FRA,France,NaN,NaN,1996-10-12,Tail
...,...,...,...,...,...,...,...,...,...,...,...
139388,142859,Guillaume Martin,1993-06-09,Paris XIVe,Paris,FRA,France,NaN,NaN,NaN,Tail
139420,142893,Antoine Rozner,1993-02-12,Paris,Paris,FRA,France,NaN,NaN,NaN,Tail
139423,142896,Dika Mem,1997-08-31,Paris,Paris,FRA,France,NaN,NaN,NaN,Tail
139450,142923,Hadrien Salvan,1997-07-10,Paris,Paris,FRA,France,NaN,NaN,NaN,Tail


Fusion par merge. par id

In [68]:
combined_df = pd.merge(results,bios,on='athlete_id', how= 'left')
combined_df

,year,type,discipline,event,as,athlete_id,noc,team,place,tied,...,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,heigt_category
0,1912.0,Summer,Tennis,"Singles, Men (Olympic)",Jean-François Blanchy,1,FRA,NaN,17.0,True,...,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Tail
1,1912.0,Summer,Tennis,"Doubles, Men (Olympic)",Jean-François Blanchy,1,FRA,Jean Montariol,NaN,False,...,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Tail
2,1920.0,Summer,Tennis,"Singles, Men (Olympic)",Jean-François Blanchy,1,FRA,NaN,32.0,True,...,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Tail
3,1920.0,Summer,Tennis,"Doubles, Mixed (Olympic)",Jean-François Blanchy,1,FRA,Jeanne Vaussard,8.0,True,...,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Tail
4,1920.0,Summer,Tennis,"Doubles, Men (Olympic)",Jean-François Blanchy,1,FRA,Jacques Brugnon,4.0,False,...,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Tail
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
308403,2022.0,Winter,Luge,"Singles, Men (Olympic)",Marián Skupek,148983,SVK,NaN,26.0,False,...,Marián Skupek,2001-07-12,Gelnica,Košice,SVK,Slovakia,196.0,108.0,NaN,Tail
308404,2022.0,Winter,Alpine Skiing (Skiing),"Slalom, Women (Olympic)",Elsa Fermbäck,148984,SWE,NaN,28.0,False,...,Elsa Fermbäck,1998-03-28,Vemdalen,Jämtland,SWE,Sweden,NaN,NaN,NaN,Tail
308405,2022.0,Winter,Alpine Skiing (Skiing),"Team, Mixed (Olympic)",Hilma Lövblom,148985,SWE,Sweden,13.0,False,...,Hilma Lövblom,2000-08-16,Täby,Stockholm,SWE,Sweden,NaN,NaN,NaN,Tail
308406,2022.0,Winter,Alpine Skiing (Skiing),"Giant Slalom, Women (Olympic)",Hilma Lövblom,148985,SWE,NaN,NaN,False,...,Hilma Lövblom,2000-08-16,Täby,Stockholm,SWE,Sweden,NaN,NaN,NaN,Tail


In [69]:
results.head()

,year,type,discipline,event,as,athlete_id,noc,team,place,tied,medal
0,1912.0,Summer,Tennis,"Singles, Men (Olympic)",Jean-François Blanchy,1,FRA,NaN,17.0,True,NaN
1,1912.0,Summer,Tennis,"Doubles, Men (Olympic)",Jean-François Blanchy,1,FRA,Jean Montariol,NaN,False,NaN
2,1920.0,Summer,Tennis,"Singles, Men (Olympic)",Jean-François Blanchy,1,FRA,NaN,32.0,True,NaN
3,1920.0,Summer,Tennis,"Doubles, Mixed (Olympic)",Jean-François Blanchy,1,FRA,Jeanne Vaussard,8.0,True,NaN
4,1920.0,Summer,Tennis,"Doubles, Men (Olympic)",Jean-François Blanchy,1,FRA,Jacques Brugnon,4.0,False,NaN


GERER LES VALEUR NUL DANS NOS DONEES.

In [70]:
coffes

,jour,Type café,Unité vendue,price,Revenue
0,Monday,Espresso,25,4.99,124.75
1,Monday,Latte,15,4.99,74.85
2,Tuesday,Espresso,30,4.99,149.70
3,Tuesday,Latte,20,4.99,99.80
4,Wednesday,Espresso,35,4.99,174.65
5,Wednesday,Latte,25,4.99,124.75
6,Thursday,Espresso,40,4.99,199.60
7,Thursday,Latte,30,4.99,149.70
8,Friday,Espresso,45,4.99,224.55
9,Friday,Latte,35,4.99,174.65


In [71]:
coffes.loc[[0,1], 'Unité vendue'] = np.nan ## Créer une valeur manquante dans la colone unité vendue
coffes

,jour,Type café,Unité vendue,price,Revenue
0,Monday,Espresso,NaN,4.99,124.75
1,Monday,Latte,NaN,4.99,74.85
2,Tuesday,Espresso,30.0,4.99,149.70
3,Tuesday,Latte,20.0,4.99,99.80
4,Wednesday,Espresso,35.0,4.99,174.65
5,Wednesday,Latte,25.0,4.99,124.75
6,Thursday,Espresso,40.0,4.99,199.60
7,Thursday,Latte,30.0,4.99,149.70
8,Friday,Espresso,45.0,4.99,224.55
9,Friday,Latte,35.0,4.99,174.65


In [72]:
coffes[coffes['Unité vendue'].notna()]###pour faire disparaitre les valeurs manquante.

,jour,Type café,Unité vendue,price,Revenue
2,Tuesday,Espresso,30.0,4.99,149.70
3,Tuesday,Latte,20.0,4.99,99.80
4,Wednesday,Espresso,35.0,4.99,174.65
5,Wednesday,Latte,25.0,4.99,124.75
6,Thursday,Espresso,40.0,4.99,199.60
7,Thursday,Latte,30.0,4.99,149.70
8,Friday,Espresso,45.0,4.99,224.55
9,Friday,Latte,35.0,4.99,174.65
10,Saturday,Espresso,45.0,4.99,224.55
11,Saturday,Latte,35.0,4.99,174.65


In [73]:
bios.head()

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,heigt_category
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Tail
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN,average
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17,average
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20,average
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25,Tail


In [74]:
bios['born_city'].value_counts() ## la plupart des personnes viennent de .. ensuite .. et ..

born_city
Budapest            1378
Moskva (Moscow)      883
Oslo                 708
Stockholm            629
Praha (Prague)       600
                    ... 
Tsivilsk               1
Roth                   1
Romoos                 1
Dva Polya Artash       1
Dulwich Hill           1
Name: count, Length: 22368, dtype: int64

In [75]:
bios[bios['born_country'] == 'USA']['born_region'].value_counts() ## la plus part des personn
#qui son au USA vivent dans california ensuite new york et illinois

born_region
California              1634
New York                 990
Illinois                 585
Pennsylvania             530
Massachusetts            530
New Jersey               381
Texas                    368
Minnesota                365
Ohio                     328
Michigan                 319
Washington               240
Florida                  235
Wisconsin                209
Colorado                 207
Connecticut              156
Indiana                  150
Oregon                   132
Georgia                  129
Virginia                 121
Maryland                 117
District of Columbia     107
Iowa                     102
Hawaiʻi                   95
Kansas                    94
Oklahoma                  93
Louisiana                 92
Utah                      91
Missouri                  91
North Carolina            86
Arizona                   83
New Hampshire             83
Vermont                   68
Mississippi               66
Alabama                   64
Te

In [76]:
coffes['Unité vendue'] = coffes['Unité vendue'].interpolate()
coffes

,jour,Type café,Unité vendue,price,Revenue
0,Monday,Espresso,NaN,4.99,124.75
1,Monday,Latte,NaN,4.99,74.85
2,Tuesday,Espresso,30.0,4.99,149.70
3,Tuesday,Latte,20.0,4.99,99.80
4,Wednesday,Espresso,35.0,4.99,174.65
5,Wednesday,Latte,25.0,4.99,124.75
6,Thursday,Espresso,40.0,4.99,199.60
7,Thursday,Latte,30.0,4.99,149.70
8,Friday,Espresso,45.0,4.99,224.55
9,Friday,Latte,35.0,4.99,174.65


In [77]:
coffes.columns

Index(['jour', 'Type café', 'Unité vendue', 'price', 'Revenue'], dtype='str')

In [78]:
coffes

,jour,Type café,Unité vendue,price,Revenue
0,Monday,Espresso,NaN,4.99,124.75
1,Monday,Latte,NaN,4.99,74.85
2,Tuesday,Espresso,30.0,4.99,149.70
3,Tuesday,Latte,20.0,4.99,99.80
4,Wednesday,Espresso,35.0,4.99,174.65
5,Wednesday,Latte,25.0,4.99,124.75
6,Thursday,Espresso,40.0,4.99,199.60
7,Thursday,Latte,30.0,4.99,149.70
8,Friday,Espresso,45.0,4.99,224.55
9,Friday,Latte,35.0,4.99,174.65


On veut savoir le type de café vendue, ( Espresso et Lattr)

In [79]:
coffes.groupby(['Type café'])['Unité vendue'].sum()
# on a vendue 240 Espresso et 180 Latte.

Type café
Espresso    240.0
Latte       180.0
Name: Unité vendue, dtype: float64

In [80]:
coffes.groupby(['Type café'])['Unité vendue'].mean()## onn a ve,due en moyenne 40 Espresso et 30 Latte

Type café
Espresso    40.0
Latte       30.0
Name: Unité vendue, dtype: float64

Voire la somme,le prix vendue et la moyenne du revenue

In [81]:
coffes.groupby(['Type café']).agg({'Unité vendue' : 'sum','price': 'mean','Revenue' : 'mean'})


,Unité vendue,price,Revenue
Type café,,,
Espresso,240.0,4.99,188.907143
Latte,180.0,4.99,139.007143


In [82]:
coffes.columns

Index(['jour', 'Type café', 'Unité vendue', 'price', 'Revenue'], dtype='str')

Evaluer les unité vendue en fonction du jour de la semaine.

In [83]:
pivot =coffes.pivot(columns= 'Type café',  index= 'jour', values= 'Revenue')
pivot

Type café,Espresso,Latte
jour,,
Friday,224.55,174.65
Monday,124.75,74.85
Saturday,224.55,174.65
Sunday,224.55,174.65
Thursday,199.60,149.70
Tuesday,149.70,99.80
Wednesday,174.65,124.75


In [84]:
pivot.loc['Monday', 'Latte']

np.float64(74.85000000000001)

In [85]:
pivot.sum(axis=1)

jour
Friday       399.2
Monday       199.6
Saturday     399.2
Sunday       399.2
Thursday     349.3
Tuesday      249.5
Wednesday    299.4
dtype: float64

Regroupons les personne par année de naissance.

In [86]:
bios.head()

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,heigt_category
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Tail
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN,average
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17,average
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20,average
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25,Tail


In [87]:
#bios['born_date'] = pd.to_datetime(bios['born_date'])
#bios['month_born'] = bios['born_date'].dt.month
#bios['year_born'] = bios['born_date'].dt.year
#bios.groupby(bios['born_date'], bios['month_born'])['name'].count().reset_index().sort_values('name', ascending =False)

In [88]:
coffes

,jour,Type café,Unité vendue,price,Revenue
0,Monday,Espresso,NaN,4.99,124.75
1,Monday,Latte,NaN,4.99,74.85
2,Tuesday,Espresso,30.0,4.99,149.70
3,Tuesday,Latte,20.0,4.99,99.80
4,Wednesday,Espresso,35.0,4.99,174.65
5,Wednesday,Latte,25.0,4.99,124.75
6,Thursday,Espresso,40.0,4.99,199.60
7,Thursday,Latte,30.0,4.99,149.70
8,Friday,Espresso,45.0,4.99,224.55
9,Friday,Latte,35.0,4.99,174.65


In [99]:
##comment savoir comment les revenues sont sont comporter ppar "
coffes["yesterday_revenue"] = coffes['Revenue'].shift(2)
coffes

,jour,Type café,Unité vendue,price,Revenue,yesterday_revenue,pct_cgange
0,Monday,Espresso,NaN,4.99,124.75,NaN,83.333333
1,Monday,Latte,NaN,4.99,74.85,NaN,75.000000
2,Tuesday,Espresso,30.0,4.99,149.70,124.75,85.714286
3,Tuesday,Latte,20.0,4.99,99.80,74.85,80.000000
4,Wednesday,Espresso,35.0,4.99,174.65,149.70,87.500000
5,Wednesday,Latte,25.0,4.99,124.75,99.80,83.333333
6,Thursday,Espresso,40.0,4.99,199.60,174.65,88.888889
7,Thursday,Latte,30.0,4.99,149.70,124.75,85.714286
8,Friday,Espresso,45.0,4.99,224.55,199.60,100.000000
9,Friday,Latte,35.0,4.99,174.65,149.70,100.000000


In [106]:
coffes['pct_gange'] = coffes['Revenue'] / coffes['yesterday_revenue']*100
coffes

,jour,Type café,Unité vendue,price,Revenue,yesterday_revenue,pct_cgange,pct_gange
0,Monday,Espresso,NaN,4.99,124.75,NaN,NaN,NaN
1,Monday,Latte,NaN,4.99,74.85,NaN,NaN,NaN
2,Tuesday,Espresso,30.0,4.99,149.70,124.75,120.000000,120.000000
3,Tuesday,Latte,20.0,4.99,99.80,74.85,133.333333,133.333333
4,Wednesday,Espresso,35.0,4.99,174.65,149.70,116.666667,116.666667
5,Wednesday,Latte,25.0,4.99,124.75,99.80,125.000000,125.000000
6,Thursday,Espresso,40.0,4.99,199.60,174.65,114.285714,114.285714
7,Thursday,Latte,30.0,4.99,149.70,124.75,120.000000,120.000000
8,Friday,Espresso,45.0,4.99,224.55,199.60,112.500000,112.500000
9,Friday,Latte,35.0,4.99,174.65,149.70,116.666667,116.666667


In [137]:
bios['height_rank'] = bios['height_cm'].rank()


In [140]:
bios['height_rank'] = bios['height_cm'].rank(ascending=False)
bios

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,heigt_category,height_rank
0,1,Jean-François Blanchy,1886-12-12,Bordeaux,Gironde,FRA,France,NaN,NaN,1960-10-02,Tail,NaN
1,2,Arnaud Boetsch,1969-04-01,Meulan,Yvelines,FRA,France,183.0,76.0,NaN,average,27597.5
2,3,Jean Borotra,1898-08-13,Biarritz,Pyrénées-Atlantiques,FRA,France,183.0,76.0,1994-07-17,average,27597.5
3,4,Jacques Brugnon,1895-05-11,Paris VIIIe,Paris,FRA,France,168.0,64.0,1978-03-20,average,83975.0
4,5,Albert Canet,1878-04-17,Wandsworth,England,GBR,France,NaN,NaN,1930-07-25,Tail,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...
145495,149222,Polina Luchnikova,2002-01-30,Serov,Sverdlovsk,RUS,ROC,167.0,61.0,NaN,average,87188.5
145496,149223,Valeriya Merkusheva,1999-09-20,Moskva (Moscow),Moskva,RUS,ROC,168.0,65.0,NaN,average,83975.0
145497,149224,Yuliya Smirnova,1998-05-08,Kotlas,Arkhangelsk,RUS,ROC,163.0,55.0,NaN,short,96305.5
145498,149225,André Foussard,1899-05-19,Niort,Deux-Sèvres,FRA,France,166.0,NaN,1986-03-18,average,89274.5


In [143]:
jour= bios.sort_values('height_rank', ascending=True) ### génerer par l'IA.
jour

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,heigt_category,height_rank
89070,89782,Yao Ming,1980-09-12,Xuhui District,Shanghai,CHN,People's Republic of China,226.0,141.0,NaN,Tail,1.0
6978,7013,Arvydas Sabonis,1964-12-19,Kaunas,Kaunas,LTU,Lithuania Soviet Union,223.0,122.0,NaN,Tail,2.5
5781,5804,Tommy Burleson,1952-02-24,Crossnore,North Carolina,USA,United States,223.0,102.0,NaN,Tail,2.5
5673,5696,Gunther Behnke,1963-01-19,Leverkusen,Nordrhein-Westfalen,GER,Germany,221.0,114.0,NaN,Tail,5.0
89075,89787,Roberto Dueñas,1975-11-01,Madrid,Madrid,ESP,Spain,221.0,137.0,NaN,Tail,5.0
...,...,...,...,...,...,...,...,...,...,...,...,...
145490,149217,Sin Ye-Chan,1995-06-13,NaN,NaN,NaN,Republic of Korea,NaN,NaN,NaN,Tail,NaN
145491,149218,Matthew Wepke,1989-12-05,NaN,NaN,NaN,Jamaica,NaN,NaN,NaN,Tail,NaN
145492,149219,Carlos García-Ordóñez,1927-04-24,La Habana (Havana),Ciudad de La Habana,CUB,Cuba,NaN,NaN,2019-11-24,Tail,NaN
145493,149220,Landysh Falyakhova,1998-08-31,Dva Polya Artash,Respublika Tatarstan,RUS,ROC,NaN,NaN,NaN,Tail,NaN


In [ ]:
bios['sum'] = bios['height_rank'].cumsum()
bios

In [141]:
bios.sort_values(['height_rank'])

,athlete_id,name,born_date,born_city,born_region,born_country,NOC,height_cm,weight_kg,died_date,heigt_category,height_rank
89070,89782,Yao Ming,1980-09-12,Xuhui District,Shanghai,CHN,People's Republic of China,226.0,141.0,NaN,Tail,1.0
6978,7013,Arvydas Sabonis,1964-12-19,Kaunas,Kaunas,LTU,Lithuania Soviet Union,223.0,122.0,NaN,Tail,2.5
5781,5804,Tommy Burleson,1952-02-24,Crossnore,North Carolina,USA,United States,223.0,102.0,NaN,Tail,2.5
5673,5696,Gunther Behnke,1963-01-19,Leverkusen,Nordrhein-Westfalen,GER,Germany,221.0,114.0,NaN,Tail,5.0
89075,89787,Roberto Dueñas,1975-11-01,Madrid,Madrid,ESP,Spain,221.0,137.0,NaN,Tail,5.0
...,...,...,...,...,...,...,...,...,...,...,...,...
145490,149217,Sin Ye-Chan,1995-06-13,NaN,NaN,NaN,Republic of Korea,NaN,NaN,NaN,Tail,NaN
145491,149218,Matthew Wepke,1989-12-05,NaN,NaN,NaN,Jamaica,NaN,NaN,NaN,Tail,NaN
145492,149219,Carlos García-Ordóñez,1927-04-24,La Habana (Havana),Ciudad de La Habana,CUB,Cuba,NaN,NaN,2019-11-24,Tail,NaN
145493,149220,Landysh Falyakhova,1998-08-31,Dva Polya Artash,Respublika Tatarstan,RUS,ROC,NaN,NaN,NaN,Tail,NaN


In [144]:
bios.sort_values(['height_rank']).sample(10)[['name', 'height_rank']]

,name,height_rank
11142,Milan Janić,46966.0
49616,Moira Abernethy,NaN
139794,Franziska Brauße,NaN
14734,Toivo Hörkkö,NaN
39105,Grzegorz Stellak,2609.0
16161,Ret Chhon,96305.5
110494,Kristina Vestveit,81027.5
134068,Marko Kilp,34234.5
25643,P. Venkatesh,NaN
22031,Edgar Seligman,62600.5
